lib/translate/lun_translator.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lun_translator.h"
+
+namespace translator {
+
+// Section 4.5
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+std::optional<nvme_defs::GenericQueueEntryCmd> translateReportLunCommand(
+    scsi_defs::ReportLunsCommand* scsi_cmd, uint64_t prp_ptr) {
+  if (scsi_cmd->select_report != scsi_defs::SelectReport::kRestrictedMethods &&
+      scsi_cmd->select_report != scsi_defs::SelectReport::kWellKnown &&
+      scsi_cmd->select_report != scsi_defs::SelectReport::kAllLogical)
+    return std::nullopt;
+
+  nvme_defs::GenericQueueEntryCmd nvme_cmd = {};
+  nvme_cmd.opc = (uint8_t)nvme_defs::AdminOpcode::kIdentify;
```

Comment:
[Use C++ casts](https://google.github.io/styleguide/cppguide.html#Casting). In this case, a `static_cast` would be appropriate.